### PR TITLE
Switched DropShadow instantiation from managed to raw pointers

### PR DESCRIPTION
**Pull request by spireken**
I put a build in the drafts/demos folder in case it's needed.

I tested the callback using the context menu; when the context menu is hidden/destroyed there's two calls to deleteLater(), one for each menu's drop shadow.

### CODE REVIEW

File: Applications/Spire/Include/Spire/Ui/DropShadow.hpp [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 namespace Spire {
 
-  //! Paints a drop shadow around its parent widget.
+  //! Paints a drop shadow around its parent widget. The drop shadow instance
+  //! is deleted when its parent is deleted.
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/DropShadow.hpp]: Don't need to specify the deletion policy, this is how all ```QObject```s behave.

File: Applications/Spire/Source/Ui/DropShadow.cpp [hunk 2]
```diff
@@ -43,6 +43,9 @@ DropShadow::DropShadow(bool is_menu_shadow, bool has_top, QWidget* parent)
   setAttribute(Qt::WA_TranslucentBackground);
   setAttribute(Qt::WA_ShowWithoutActivating);
   m_parent->window()->installEventFilter(this);
+  connect(m_parent,
+    static_cast<void (QObject::*)(QObject*)>(&QObject::destroyed),
```
Comment (kranar) [inline Applications/Spire/Source/Ui/DropShadow.cpp]: Just to confirm, the following didn't work?

    connect(m_parent, &QObject::destroyed, ...);

Comment (spireken) [reply Applications/Spire/Source/Ui/DropShadow.cpp]: No, the compiler complained something to the effect of 'this doesn't evaluate to an expression taking one parameter'.

Comment (kranar) [reply Applications/Spire/Source/Ui/DropShadow.cpp]: Good, use ```qOverload``` from now on:

    connect(m_parent, qOverload<>(QObject::destroyed), ...);

**Comment by spireken** [comment]
Just verified it. The line has to be broken up (deleteLater() on its own line) in VS otherwise it only triggers the breakpoint on construction.

**Pull request merged**